### DISCUSSION TITLE
Reconstruct model from JSON schema

### DISCUSSION THREAD

**Discussion by Cyberwizzard** [pydantic/pydantic#10801] [Question]
To share a model across some scripts which communicate through a database, I am searching for a way to use the generated JSON schema to reconstruct a model.

The use case here is that if the schema changes, without downtime we could instantiate the new model from the JSON schema and switch validation on the fly.

But although I can find examples on how to get the JSON, I can not find an example how to load it back into a model.

Is this possible somehow?

**Comment by bruno-f-cruz** [comment]
If I understand correctly, you may want to look into the featured [datamodel_code_generator]( https://docs.pydantic.dev/latest/integrations/datamodel_code_generator/) tool.

**Comment by Cyberwizzard** [comment]
I found that one as well, but how would I use that tool to load a new
version of a model while running a script?

The description seems to indicate this is to generate a new Python file.
This sounds more like something done before starting a script and not to
dynamically update or replace a model?

**Comment by in-op** [comment]
I'm facing the same problem. Everyone points to datamodel_code_generator, but it's a CLI tool to generate code, not a library we can use to generate a pydantic model at runtime. AFAICT this doesn't exist...
